### DISCUSSION TITLE
Will test case grading consider GameState/Gameboard dependency?

### DISCUSSION THREAD

**Discussion by DelusionArun** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#12] [Q&A] [answered]
It appears that many test cases of GameState.java/Gameboard.java are dependant on Gameboard initialization. So, if there is wrong exception in AllGemsReachable, these test cases do not get implemented.

For instance, even if the implementation for all cases in GameState is correct, a mistake in flood-fill implementation of GameBoard.java will cause most cases to fail (because of generated exception). But if we don't write a flood-fill algorithm, it will work just fine.

I think the grading could be problematic in such dependency because not implementing AllGemsReachable would lead to 1 mistake (only in GameBoard) but if we did AllGemsReachable and it had a minor mistake then we would have 11-13 mistake.

So, I wonder if default Gameboard is used to check GameState.java while grading? or will the cases be run directly to our submitted folder?

**Comment by Derppening** [comment, marked as answer]
I understand your concern. During the final grading, there will be several grading strategies in place that will help prevent this kind of "double-jeopardy" point deduction from happening.

To answer your question, I will use your GameBoard to test the implementation of everything else. However, as I mentioned, I will also manually review each assignments' grading if necessary. You will also be given the opportunity to appeal your grades if you do not think that there has been an error with the grading.

I will explain how this works in detail when the assignment grades and the solution is released.

**Reply by DelusionArun** [reply]
Thank you for clarifying.
